src/palette/internal/palette.cpp

```diff
@@ -515,102 +508,143 @@ Palette::Type Palette::guessType() const
     case ElementType::FERMATA:
-        return Type::Articulation;
+        return Palette::Type::Articulation;
     case ElementType::BREATH:
-        return Type::Breath;
+        return Palette::Type::Breath;
     case ElementType::NOTEHEAD:
-        return Type::NoteHead;
+        return Palette::Type::NoteHead;
     case ElementType::BAR_LINE:
-        return Type::BarLine;
+        return Palette::Type::BarLine;
     case ElementType::ARPEGGIO:
     case ElementType::GLISSANDO:
-        return Type::Arpeggio;
+        return Palette::Type::Arpeggio;
     case ElementType::TREMOLO:
-        return Type::Tremolo;
+        return Palette::Type::Tremolo;
     case ElementType::TEMPO_TEXT:
-        return Type::Tempo;
+        return Palette::Type::Tempo;
     case ElementType::DYNAMIC:
-        return Type::Dynamic;
+        return Palette::Type::Dynamic;
     case ElementType::FINGERING:
-        return Type::Fingering;
+        return Palette::Type::Fingering;
     case ElementType::MARKER:
     case ElementType::JUMP:
     case ElementType::MEASURE_REPEAT:
-        return Type::Repeat;
+        return Palette::Type::Repeat;
     case ElementType::FRET_DIAGRAM:
-        return Type::FretboardDiagram;
+        return Palette::Type::FretboardDiagram;
     case ElementType::BAGPIPE_EMBELLISHMENT:
-        return Type::BagpipeEmbellishment;
+        return Palette::Type::BagpipeEmbellishment;
     case ElementType::LAYOUT_BREAK:
     case ElementType::SPACER:
-        return Type::Layout;
+        return Palette::Type::Layout;
     case ElementType::SYMBOL:
-        return Type::Accordion;
+        return Palette::Type::Accordion;
     case ElementType::ACTION_ICON: {
         const ActionIcon* action = toActionIcon(e);
         QString actionCode = QString::fromStdString(action->actionCode());
 
         if (actionCode.contains("beam")) {
-            return Type::Beam;
+            return Palette::Type::Beam;
         }
         if (actionCode.contains("grace") || actionCode.contains("acciaccatura") || actionCode.contains("appoggiatura")) {
-            return Type::GraceNote;
+            return Palette::Type::GraceNote;
         }
         if (actionCode.contains("frame") || actionCode.contains("box") || actionCode.contains("measure")) {
-            return Type::Layout;
+            return Palette::Type::Layout;
         }
-        return Type::Custom;
+        return Palette::Type::Custom;
     }
     default: {
         if (e->isSpanner()) {
-            return Type::Line;
+            return Palette::Type::Line;
         }
         if (e->isTextBase()) {
-            return Type::Text;
+            return Palette::Type::Text;
         }
     }
     }
 
-    return Type::Custom;
+    return Palette::Type::Custom;
 }
 
-std::function<void(PaletteCellPtr)> Palette::cellHandlerByPaletteType(const Palette::Type& type) const
+void scaleCell(PaletteCellPtr cellPtr)
```

Comment:
No need to return a lambda; we can just operate on a `PaletteCellPtr` directly! :relaxed: